earthfile2llb/converter.go

```diff
@@ -527,7 +527,7 @@ func (c *Converter) RunExitCode(ctx context.Context, opts ConvertRunOpts) (int,
 			return os.RemoveAll(exitCodeDir)
 		})
 	} else {
-		exitCodeFile = "/run/exit_code"
+		exitCodeFile = "/tmp/earthly_if_statement_exit_code"
```

Comment:
IIRC, we use `/run/` for other things too. Mind double-checking that this is the only spot?

Reply:
I was able to fix `ARG` too.

However, I see that `WITH DOCKER` doesn't work -- which makes a bit of sense since docker typically requires root access (or for the user to be part of the docker group, which doesn't exist under our DIND image).

I additionally created #1549 for creating ssh auth sockets which isn't a quick fix.